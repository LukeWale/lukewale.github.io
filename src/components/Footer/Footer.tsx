import React from "react";
import './Footer.css';

const Footer = () => {
    return (
        <footer data-testid="app-footer" className="bg-gray-800 p-6 sticky top-[100vh] text-white">
            <div className="container grid grid-cols-3">
                <nav role="nav">
                    <div data-testid="nav-content" id="nav-content" className={`w-full flex-grow lg:flex lg:items-center lg:w-auto lg:block pt-6 lg:pt-0`}>
                        <ul className="list-reset lg:flex justify-end flex-1 flex-col">
                            <li className="mr-3">
                                <a className="inline-block py-2 px-4 text-white no-underline" href="#!">Home</a>
                            </li>
                            <li className="mr-3">
                                <a className="inline-block py-2 px-4 text-white no-underline" href="#!">GitHub</a>
                            </li>
                            <li className="mr-3">
                                <a className="inline-block py-2 px-4 text-white no-underline" href="#!">Contact</a>
                            </li>
                        </ul>
                    </div>
                </nav>
            </div>
            
            <div className="text-center py-3">
                <p>Copyright Luke Wale</p>
            </div>
        </footer>
    );
    
}

export default Footer;