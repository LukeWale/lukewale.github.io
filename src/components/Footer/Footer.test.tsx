import React from "react";
import { cleanup, render, screen, within } from '@testing-library/react';
import Footer from './Footer';

describe('Footer', () => {
    afterEach(cleanup);

    it('should render correctly', () => {
        const { asFragment } = render(<Footer />);

        expect(asFragment()).toMatchSnapshot();
    })

    it('should have copyright text', () => {
        render(<Footer />);

        const footer = screen.getByTestId('app-footer');

        within(footer).getByText(/Copyright Luke Wale/)
    });

    it('should have a navigation', () => {
        render(<Footer />);

        const footerNav = screen.getByRole('nav');

        expect(footerNav).toBeInTheDocument();

    });

    it('navigation should have an unordered list with at least one link', () => {
        render(<Footer />);

        const footerNav = screen.getByRole('nav');

        within(footerNav).getByRole('list');
        within(footerNav).getAllByRole('listitem');

    });

    it('Has a link to the home page', () => {
        render(<Footer />);

        const menu = screen.getByTestId('nav-content');

        expect(menu).toBeInTheDocument();
        within(menu).getByText(/Home/);
        
    });

    it('Has a link to my github', () => {
        render(<Footer />);

        const menu = screen.getByTestId('nav-content');

        expect(menu).toBeInTheDocument();
        within(menu).getByText(/GitHub/);
    });

    it('Has a link to contact me', () => {
        render(<Footer />);

        const menu = screen.getByTestId('nav-content');

        expect(menu).toBeInTheDocument();
        within(menu).getByText(/Contact/);
    });


});