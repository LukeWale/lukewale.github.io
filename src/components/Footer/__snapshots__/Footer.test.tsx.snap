// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly 1`] = `
<DocumentFragment>
  <footer
    class="bg-gray-800 p-6 sticky top-[100vh] text-white"
    data-testid="app-footer"
  >
    <div
      class="container grid grid-cols-3"
    >
      <nav
        role="nav"
      >
        <div
          class="w-full flex-grow lg:flex lg:items-center lg:w-auto lg:block pt-6 lg:pt-0"
          data-testid="nav-content"
          id="nav-content"
        >
          <ul
            class="list-reset lg:flex justify-end flex-1 flex-col"
          >
            <li
              class="mr-3"
            >
              <a
                class="inline-block py-2 px-4 text-white no-underline"
                href="#!"
              >
                Home
              </a>
            </li>
            <li
              class="mr-3"
            >
              <a
                class="inline-block py-2 px-4 text-white no-underline"
                href="#!"
              >
                GitHub
              </a>
            </li>
            <li
              class="mr-3"
            >
              <a
                class="inline-block py-2 px-4 text-white no-underline"
                href="#!"
              >
                Contact
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
    <div
      class="text-center py-3"
    >
      <p>
        Copyright Luke Wale
      </p>
    </div>
  </footer>
</DocumentFragment>
`;
